kratos/spatial_containers/search_wrapper.cpp

```diff
@@ -0,0 +1,334 @@
+            DistanceVector results_distances(AllocationSize);
+            const std::size_t number_of_results = mpSearchObject->SearchInRadius(rPoint, Radius, results.begin(), results_distances.begin(), AllocationSize);
+            if (number_of_results > 0) {
+                // Resize the results
+                results_distances.resize(number_of_results);
+
+                // Find the iterator pointing to the smallest value
+                auto it_min = std::min_element(results_distances.begin(), results_distances.end());
+
+                // Calculate the index
+                IndexType index = std::distance(results_distances.begin(), it_min);
+
+                // Set the result
+                rResult = ResultType((results[index]->pGetObject()).get(), rank);
+                rResult.SetDistance(results_distances[index]);
+            }
+        }
+    }
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<class TSearchObject>
+void SearchWrapper<TSearchObject>::LocalSearchNearest(
+    const PointType& rPoint,
+    ResultType& rResult
+    )
+{
+    // Get the rank
+    const int rank = GetRank();
+
+    // If we are using GeometricalObjectBins we can use the optimized search
+    if constexpr (IsGeometricalObjectBins) {
+        rResult = mpSearchObject->SearchNearest(rPoint);
+        rResult.Get().SetRank(rank);
+    } else { // Using trees
+        // We search if geometrical objects are provided
+        if (mpPointVector) {
+            // Search nearest
+            double distance = 0.0;
+            auto p_point = mpSearchObject->SearchNearestPoint(rPoint, distance);
+
+            // Set the result
+            rResult = ResultType((p_point->pGetObject()).get(), rank);
+            rResult.SetDistance(distance);
+        }
+    }
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<class TSearchObject>
+void SearchWrapper<TSearchObject>::LocalSearchIsInside(
+    const PointType& rPoint,
+    ResultType& rResult
+    )
+{
+    // If we are using GeometricalObjectBins we can use the optimized search
+    if constexpr (IsGeometricalObjectBins) {
+        rResult = mpSearchObject->SearchIsInside(rPoint);
+        rResult.Get().SetRank(GetRank());
+    } else { // Using trees
+        KRATOS_ERROR << "SearchIsInside not compatible with Search trees" << std::endl;
+    }
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<class TSearchObject>
+const Parameters SearchWrapper<TSearchObject>::GetDefaultParameters() const 
+{
+    const Parameters default_parameters = Parameters(R"(
+    {
+        "allocation_size" : 1000,
```

Comment:
This is the way the trees work, you need to provide an allocation size manually. I can add an automatic guess instead of manually assign, or by default -1 as value, so the value is guessed, and to consider a specific value if explicitily provided.